elasticai/creator/nn/quantized_grads/base_modules/batchnorm2d.py

```diff
@@ -1,58 +1,22 @@
-from typing import Any, Callable, Protocol
+from typing import Any
 
 from torch import Tensor
-from torch.nn import BatchNorm2d as __BatchNorm2d
+from torch.nn import BatchNorm2d as TorchBatchNorm2d
+from torch.nn import Module
+from torch.nn.utils import parametrize as P
 
-from elasticai.creator.base_modules.math_operations import Quantize
-from elasticai.creator.nn.quantized_grads.quantized_parameters import (
-    QuantizationSchemeByName,
-    QuantizedParameters,
-)
 
-
-class _BatchNorm2d(__BatchNorm2d):
-    """
-    This class is a facade for batchnorm to allow clean multiple inheritance for BatchNorm2d.
-    """
-
-    def __init__(
-        self,
-        num_features: int,
-        eps: float = 1e-5,
-        momentum: float = 0.1,
-        affine: bool = True,
-        track_running_stats: bool = True,
-        device: Any = None,
-        dtype: Any = None,
-        *args,
-        **kwargs,
-    ):
-        super().__init__(
-            num_features=num_features,
-            eps=eps,
-            momentum=momentum,
-            affine=affine,
-            track_running_stats=track_running_stats,
-            device=device,
-            dtype=dtype,
-        )
-
-
-class MathOperations(Quantize, Protocol): ...
-
-
-class BatchNorm2d(_BatchNorm2d, QuantizedParameters):
+class BatchNorm2d(TorchBatchNorm2d):
     """This module implements a 2d batch norm.
     The output of the batchnorm is fake quantized. The weights and bias are fake quantized during initialization.
     To keep the weights quantized use only optimizers that apply a quantized update.
-    IMPORTANT: use only in-place operators for your quantize implementations
     """
 
     def __init__(
         self,
-        operations: MathOperations,
-        weight_quantization: Callable[[Tensor], None],
-        bias_quantization: Callable[[Tensor], None],
+        math_ops: Module,
```

Comment:
also referring to your question, if it serves the same purpose as `MathOperations` and it just happens to additionally be a module, i'd just stick to calling it operations here.

Reply:
A very brief comment explaining why this is a module here, will probably be useful in the future